a-rmz/JEasyCrypto
# Feature gui

#### What does this PR do?
This PR adds a graphical user interface to the system. The first commits separate the network logic from the ui logic. Then the gui classes are added. The last commits fix the build scripts for the entire system

#### Where should the reviewer start?
The reviews should start with the CryptoClient class and should check the new method signatures. Than the ConsoleLauncher class should be checked to ensure that the console user interface still works properly. Finally the reviews should analyze the gui classes.

#### How should this be manually tested?
1. All methods should be tested from the console interface (use script ./start.sh to start the client)
2. All methods should be tested from graphical interface (use script ./startGui.sh to start the client)

#### Any background context you want to provide?
The graphical user interface uses JavaFX with fxml files

#### What are the relevant tickets?
Closes #16 

#### Screenshots
![screenshot from 2017-10-19 22-25-25](https://user-images.githubusercontent.com/31992665/31790035-c6175882-b51c-11e7-8f75-d1c95b010f9a.png)
![screenshot from 2017-10-19 22-25-15](https://user-images.githubusercontent.com/31992665/31790036-c635f800-b51c-11e7-8e85-81aa08a09b93.png)
